File: solutions/javascript/strain/1/strain.js

export const keep = (collection, predicate) => {
  const result = [];
  
  for (const item of collection) {
    if (predicate(item)) {
      result.push(item);
    }
  }
  
  return result;
};


export const discard = (collection, predicate) => {
  const result = [];
  
  for (const item of collection) {
    if (!predicate(item)) {
      result.push(item);
    }
  }
  
  return result;
};

File: solutions/julia/strain/1/strain.jl
function keep(values, predicate)
    result = []
    for x in values
        if predicate(x)
            push!(result, x)
        end
    end
    return result
end

function discard(values, predicate)
    result = []
    for x in values
        if !predicate(x)
            push!(result, x)
        end
    end
    return result
end
